core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariClusterConnector.java

```diff
@@ -448,11 +448,12 @@ public boolean deleteHostFromAmbari(Stack stack, HostMetadata data) throws Cloud
         return hostDeleted;
     }
 
-    private String extendBlueprintWithFsConfig(String blueprintText, FileSystem fs) throws IOException {
+    private String extendBlueprintWithFsConfig(String blueprintText, FileSystem fs, Stack stack) throws IOException {
         FileSystemConfigurator fsConfigurator = fileSystemConfigurators.get(FileSystemType.valueOf(fs.getType()));
         ObjectMapper mapper = new ObjectMapper();
         String json = mapper.writeValueAsString(fs.getProperties());
         FileSystemConfiguration fsConfiguration = (FileSystemConfiguration) mapper.readValue(json, FileSystemType.valueOf(fs.getType()).getClazz());
+        fsConfiguration.addProperty(FileSystemConfiguration.STORAGE_CONTAINER, "cloudbreak" + stack.getId());
         List<BlueprintConfigurationEntry> bpConfigEntries = fsConfigurator.getFsProperties(fsConfiguration);
         if (fs.isDefaultFs()) {
             bpConfigEntries.addAll(fsConfigurator.getDefaultFsProperties(fsConfiguration));
```

Comment:
Can you change it to provide only an abstract parameter like `containerName` here so this interface remains independent of the domain model? Maybe we can even use filesystem name here, it's not used for anything else.